File: media3/media3.py
n1 = float(input())
n2 = float(input())
n3 = float(input())
n4 = float(input())

media = (2*n1 + 3*n2 + 4*n3 + 1*n4)/10

print(f"Media: {media}")

if(media >= 7.0):
    print("Aluno aprovado.")
elif(media >= 5.0 and media <= 6.9):
    print("Aluno em exame.")
    exame = float(input())
    print(f"Nota do exame: {exame}")
    media_final = (media+exame)/2
    if(media_final >= 5.0):
        print("Aluno aprovado.")
        print(f"Media final: {media_final}")
    else:
        print("Aluno reprovado.")
        print(f"Media final: {media_final}")
else:
    print("Aluno reprovado")

File: media3/media3.c
#include <stdio.h>
 
int main() {
 
    float n1,n2,n3,n4, exame;
    scanf("%f %f %f %f", &n1,&n2,&n3,&n4);
    float media = ((2*n1) + (3*n2) + (4*n3) + (1*n4))/10;
    printf("Media: %.1f\n", media);
    
    if(media >= 7.0){
        printf("Aluno aprovado.\n");
    } else if (media < 5.0){
        printf("Aluno reprovado.\n");
    } else if (media >= 5.0 && media <= 6.9){
        printf("Aluno em exame.\n");
        scanf("%f", &exame);
        printf("Nota do exame: %.1f\n", exame);
        float media_final = (media + exame)/2;
        if(media_final >= 5.0){
            printf("Aluno aprovado.\n");
            printf("Media final: %.1f\n", media_final);
        } else {
            printf("Aluno reprovado.\n");
            printf("Media final: %.1f\n", media_final);
        }
    }
 
    return 0;
}